src/gs-plugin-apk/gs-plugin-apk.c

```diff
@@ -685,20 +709,33 @@ resolve_matching_package (GsPlugin *plugin,
   return TRUE;
 }
 
-gboolean
-gs_plugin_refine (GsPlugin *plugin,
-                  GsAppList *apps,
-                  GsPluginRefineFlags flags,
-                  GCancellable *cancellable,
-                  GError **error)
+static gboolean
+gs_plugin_apk_refine_finish (GsPlugin      *plugin,
+                             GAsyncResult  *result,
+                             GError       **error)
 {
-  g_autoptr (GError) local_error = NULL;
+  return g_task_propagate_boolean (G_TASK (result), error);
+}
+
+static void
+gs_plugin_apk_refine_async (GsPlugin            *plugin,
+                            GsAppList           *list,
+                            GsPluginRefineFlags  flags,
+                            GCancellable        *cancellable,
+                            GAsyncReadyCallback  callback,
+                            gpointer             user_data)
+{
+  g_autoptr (GTask) task = NULL;
+  g_autoptr (GError) error = NULL;
+
+  task = g_task_new (plugin, cancellable, callback, user_data);
+  g_task_set_source_tag (task, gs_plugin_apk_refine_async);
 
```

Comment:
Here we need to test against the `NONE` flag. As we should totally skip the refine if that is the only flag set.

```C
if (flags == GS_PLUGIN_REFINE_FLAGS_NONE)
  return TRUE;
```